codegen/config/binder.go

```diff
@@ -172,28 +174,34 @@ type TypeReference struct {
 	Definition  *ast.Definition
 	GQL         *ast.Type
 	GO          types.Type
+	BaseType    types.Type  // Before calling marshalling functions cast from/to this base type
 	Marshaler   *types.Func // When using external marshalling functions this will point to the Marshal function
 	Unmarshaler *types.Func // When using external marshalling functions this will point to the Unmarshal function
+	IsMarshaler bool        // Does the type implement graphql.Marshaler and graphql.Unmarshaler
```

Comment:
Could you maybe wrap `IsMarshaler` and `BaseType` into a struct or something to indicate they're only set together?  The names seems a bit generic and since `TypeReference` is used everywhere I can imagine being confused by this field in the future.

Reply:
They are mutually exclusive options, builtins never implement Marshal.

Open to naming suggestions, naming is hard. Its the underlying type but only when the type cant be marshalled and is currently only set to string.